Source/UnitTests/Core/PowerPC/JitArm64/MovI2R.cpp

```diff
@@ -79,6 +79,38 @@ TEST(JitArm64, MovI2R_Rand)
   }
 }
 
+// Construct and test every 64-bit logical immediate
+TEST(JitArm64, MovI2R_LogImm)
+{
+  TestMovI2R test;
+
+  u64 imm;
+  for (unsigned size = 2; size <= 64; size *= 2)
+  {
+    for (unsigned length = 1; length < size; ++length)
+    {
+      imm = ~u64{0} >> (64 - length);
```

Comment:
```suggestion
      u64 imm = ~u64{0} >> (64 - length);
```

Instead of declaring the variable before the loop.